docker/Dockerfile.dev

```diff
@@ -0,0 +1,41 @@
+# syntax=docker/dockerfile:1
+
+FROM ubuntu:22.04
+
+# Install packages as root
+# Install bazel
+RUN apt update && apt upgrade -y
+RUN apt install apt-transport-https curl gnupg -y
+RUN curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel-archive-keyring.gpg
+RUN mv bazel-archive-keyring.gpg /usr/share/keyrings
+RUN echo "deb [arch=amd64 signed-by=/usr/share/keyrings/bazel-archive-keyring.gpg] https://storage.googleapis.com/bazel-apt stable jdk1.8" | tee /etc/apt/sources.list.d/bazel.list
+RUN apt update 
+RUN apt install bazel-7.2.1 -y
+RUN ln -s /usr/bin/bazel-7.2.1 /usr/bin/bazel
+
+# install gcloud
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN apt update 
+RUN apt install google-cloud-cli -y
+
+# Install git
+RUN apt install git -y
+
+# Install podman docker
+RUN apt install podman-docker -y
+
+# Create gcloud config directories in /tmp
+RUN mkdir -p /tmp/.config/gcloud
+RUN chmod -R 777 /tmp/.config
+
+# copy user-bazelrc into /tmp for use during build
+COPY docker/user-bazelrc /tmp/.user-bazelrc
+
+# copy .bazelrc into root's home dir
+COPY docker/bazelrc /tmp/.bazelrc
+
+# set up builduser UID of 1016:1017
+# name is required for bazel to run the build
+RUN groupadd --gid 1017 buildgroup
+RUN useradd --gid buildgroup --uid 1016 -ms /bin/bash builduser
```

Comment:
I'm not sure if hardcoded UID and GID are the best options here. We could approve for now as it will work with our CI, but should open an Issue to change to be more generic.

Reply:
Yeah, I'm not thrilled about the way I did it, but wasn't sure how to get around it. Normally, it wouldn't be an issue, but bazel actually needs the user name of the account doing the building or else it bombs out. Other than that, we wouldn't care.